addons/point_of_sale/models/pos_order.py

```diff
@@ -479,6 +479,18 @@ def _get_rounded_amount(self, amount):
         currency = self.currency_id
         return currency.round(amount) if currency else amount
 
+    def get_partner_bank_id(self):
+        bank_partner_id = False
+        for id in self.payment_ids.payment_method_id:
```

Comment:
by looping on it you'll change the value each time, and finaly only get what the last payment has as value.

What you can do is:

has_pay_later = any(not pm.journal_id for pm in self.payment_ids.mapped(payment_method_id)
if has_pay_later:
    if self.amount_total <= 0 and self.partner_id.bank_ids:
        bank_partner_id = self.partner_id.bank_ids[0].id
    elif self.amount_total >= 0 and self.company_id.partner_id.bank_ids:
        bank_partner_id = self.company_id.partner_id.bank_ids[0].id
    else:
        bank_partner_id = False
else:
    bank_partner_id = False